docs/usage/topology_aware_routing.md

```diff
@@ -0,0 +1,93 @@
+- TopologyAwareHints does not work at low-endpoint counts.
+- Hints provided by the EndpointSlice controller are not deterministic. With cluster-autoscaler running and load increasing in the next moment hints can be removed. There is no option to enforce the zone-level topology.
+
+For more details, see the following issue [kubernetes/kubernetes#113731](https://github.com/kubernetes/kubernetes/issues/113731).
+
+To circumvent these issues with the EndpointSlice controller, a mutating webhook in the resource-manager assigns hints to EndpointSlice resources. For each enpoint in the EndpointSlice it sets the endpoint's hints to the endpoint's zone. The webhook overwrites the hints provided by the EndpointSlice controller in kube-controller-manager. For more details, see the [webhook's documentation](../concepts/resource-manager.md#endpointslice-hints).
+
+##### kube-proxy
+
+By default, with kube-proxy running in `iptables` mode traffic is distributed randomly across all endpoints, regardless of where it originates from. In a 3 zone cluster, traffic is more likely to go to another zone than to stay in the current zone.
+With the `TopologyAwareRouting` feature kube-proxy filters the endpoints it routes to based on the hints in the EndpointSlice resource. In most of the cases, kube-proxy will prefer the endpoint(s) in the same zone. For more details, see the [following section](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/#implementation-kube-proxy).
+
+## Topology-aware Services in the Seed cluster
+
+##### etcd-main-client and etcd-events-client
+
+The `etcd-main-client` and `etcd-events-client` Services are topology-aware. They are consumed by the kube-apiserver.
+
+##### kube-apiserver
+
+The `kube-apiserver` Service is topology-aware. It is consumed by the controllers running in the Shoot control plane.
+
+##### gardener-resource-manager
+
+The `gardener-resource-manager` Service that is part of the Shoot control plane is topology-aware. The resource-manager serves webhooks and the Service is consumed by the kube-apiserver for the webhook communication.
+
+##### vpa-webhook
+
+The `vpa-webhook` Service that is part of the Shoot control plane is topology-aware. It is consumed by the kube-apiserver for the webhook communication.
+
+## Topology-aware Services in the Garden cluster
+
+TBD
+
+## How to make a Service topology-aware?
+
+To make a Service topology-aware the following annotation and label has to be added to the Service:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    service.kubernetes.io/topology-aware-hints: "auto"
+  labels:
+    endpoint-slice-hints.resources.gardener.cloud/consider: "true"
```

Comment:
Is there a reason to use label and annotation combination, can we not use only annotation or only labels?

Reply:
As described in https://github.com/gardener/gardener/pull/7191#discussion_r1081179150, the `service.kubernetes.io/topology-aware-hints="auto"` annotation is needed for kube-proxy. The "endpoint-slice-hints.resources.gardener.cloud/consider=true" label is needed for GRM's EndpointSlice hints mutating webhook to select only the EndpointSlice resources with the consider label and not all the EndpointSlice resources in the cluster.
I added few sentences to clarify this.